service/create/service.go

```diff
@@ -462,27 +462,42 @@ func validateIDs(ids []string) bool {
 }
 
 func (s *Service) addFunc(obj interface{}) {
-	cluster := *obj.(*awstpr.CustomObject)
+	customObject, ok := obj.(*awstpr.CustomObject)
+	if !ok {
+		s.logger.Log("error", "could not convert to awstpr.CustomObject")
+		return
+	}
+	cluster := *customObject
+
 	s.logger.Log("info", fmt.Sprintf("creating cluster '%s'", key.ClusterID(cluster)))
 
 	if err := validateCluster(cluster); err != nil {
 		s.logger.Log("error", "cluster spec is invalid: '%#v'", err)
 		return
 	}
 
-	if err := s.createClusterNamespace(cluster.Spec.Cluster); err != nil {
-		s.logger.Log("error", fmt.Sprintf("could not create cluster namespace: '%#v'", err))
+	err := s.processCluster(cluster)
+	if err != nil {
+		s.logger.Log("error", fmt.Sprintf("error processing cluster '%s': '%#v'", key.ClusterID(cluster), err))
 		return
 	}
 
+	s.logger.Log("info", fmt.Sprintf("cluster '%s' processed", key.ClusterID(cluster)))
+}
+
+func (s *Service) processCluster(cluster awstpr.CustomObject) error {
+	// Create cluster namespace in k8s.
+	if err := s.createClusterNamespace(cluster.Spec.Cluster); err != nil {
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not create cluster namespace: '%#v'", err))
+	}
```

Comment:
Yes, I agree the cluster namespace could be a separate resource. With a legacy resource for the rest of the AWS logic.